Add documentation for the UBI plugin (#7284)
opensearch-project/documentation-website: _search-plugins/ubi/dsl-queries.md

---
layout: default

> title: Example UBI queries

Reviewer: ```suggestion
title: Example UBI query DSL queries
```

> title: Sample UBI using OpenSearch DSL queries

Reviewer: ```suggestion
title: Example UBI queries
```

> parent: User behavior insights

Reviewer: ```suggestion
parent: User Behavior Insights
```

has_children: false

> nav_order: 8

Reviewer: ```suggestion
nav_order: 15
```

---

> # Example UBI queries

Reviewer: ```suggestion
# Example UBI query DSL queries
```

> # Sample UBI using OpenSearch DSL queries

Reviewer: ```suggestion
# Example UBI queries
```

> The following example UBI query uses the OpenSearch [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/). 

Reviewer: ```suggestion
You can use the OpenSearch search query language, [query DSL,]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/), to write queries against UBI data. The following example returns the number of times each `action_name` event occurs.
```

> You can use the OpenSearch search query language, [query DSL,]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/), to write queries against UBI data. The following example returns the number of times each `action_name` event occurs.

Reviewer: ```suggestion
You can use the OpenSearch search query language, [query DSL]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/), to write UBI queries. The following example returns the number of times each `action_name` event occurs.
```

For more extensive analytic queries, see [SQL queries ]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 

> 

Reviewer: #### Example request

> For more extensive analytic queries, see [SQL queries ]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 

Reviewer: ```suggestion
For more extensive analytic queries, see [Sample UBI SQL queries ]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 
```

> 

Reviewer: ```suggestion
#### Example request
```

> For more extensive analytic queries, see [Sample UBI SQL queries]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 

Reviewer: ```suggestion
For more extensive analytic queries, see [Example UBI SQL queries]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 
```

> For more extensive analytic queries, see [Sample UBI SQL queries ]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 

Reviewer: ```suggestion
For more extensive analytic queries, see [Sample UBI SQL queries]({{site.url}}{{site.baseurl}}/search-plugins/ubi/sql-queries/). 
```

```json
GET ubi_events/_search
{
  "size":0, 
  "aggs":{ 
    "event_types":{
      "terms": {
        "field":"action_name", 
        "size":10
      }
    }
  }
}
```
{% include copy.html %}

> The response contains the number of times each `action_name` event occurred.

Reviewer: ```suggestion
#### Example response
```

    "failed": 0
  },
  "hits": {
    "total": {
      "value": 10000,
      "relation": "gte"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "event_types": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "brand_filter",
          "doc_count": 3084
        },
        {
          "key": "product_hover",
          "doc_count": 3068
        },
        {
          "key": "button_click",
          "doc_count": 3054
        },
        {
          "key": "product_sort",
          "doc_count": 3012
        },
        {
          "key": "on_search",
          "doc_count": 3010
        },
        {
          "key": "type_filter",
          "doc_count": 2925
        },
        {
          "key": "login",
          "doc_count": 2433
        },
        {
          "key": "logout",
          "doc_count": 1447
        },
        {
          "key": "new_user_entry",
          "doc_count": 207
        }
      ]
    }
  }
}
```
{% include copy.html %}

You can run the preceding queries in the OpenSearch Dashboards [Query Workbench]({{site.url}}{{site.baseurl}}/search-plugins/sql/workbench/). 

> A demo workbench with data can be found here: 

Author: It was the same for @kolchfa-aws.  @epugh can you think of any way we could get around Amazon's firewall policy from our side?

> A demo workbench with data can be found here: 

Reviewer: ```suggestion
A demo workbench with sample data can be found here: 
```

  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 10000,
      "relation": "gte"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "event_types": {
      "doc_count_error_upper_bound": 0,
      "sum_other_doc_count": 0,
      "buckets": [
        {
          "key": "brand_filter",
          "doc_count": 3084
        },
        {
          "key": "product_hover",
          "doc_count": 3068
        },
        {
          "key": "button_click",
          "doc_count": 3054
        },
        {
          "key": "product_sort",
          "doc_count": 3012
        },
        {
          "key": "on_search",
          "doc_count": 3010
        },
        {
          "key": "type_filter",
          "doc_count": 2925
        },
        {
          "key": "login",
          "doc_count": 2433
        },
        {
          "key": "logout",
          "doc_count": 1447
        },
        {
          "key": "new_user_entry",
          "doc_count": 207
        }
      ]
    }
  }
}
```
{% include copy.html %}

These can be performed on the OpenSearch Dashboards/Query Workbench: 

> [http://chorus-opensearch-edition.dev.o19s.com:5601/app/OpenSearch-query-workbench](http://chorus-OpenSearch-edition.dev.o19s.com:5601/app/OpenSearch-query-workbench)

Reviewer 2: Is this link accessible? Why not link to the existing documentation of query workbench using {{site.url}}{{site.baseurl}}? Please use the link in the above sentence like `OpenSearch Dashboards [Query Workbench](link)`.
